include/dlaf/common/assert.h

```diff
@@ -24,17 +24,17 @@
 /// Message composition is lazily evaluated and it will not add any overhead if the condition is true.
 ///
 /// **This check cannot be disabled**
-#define DLAF_CHECK_WITH_ORIGIN(category, origin, condition, ...)                    \
-  if (!(condition)) {                                                               \
-    std::cerr << "[ERROR] " << origin << std::endl                                  \
-              << dlaf::common::concat(#condition, ' ', ##__VA_ARGS__) << std::endl; \
-    std::terminate();                                                               \
+#define DLAF_CHECK_WITH_ORIGIN(category, origin, condition, ...)     \
+  if (!(condition)) {                                                \
+    std::cerr << "[ERROR] " << origin << std::endl                   \
+              << dlaf::common::concat(#condition, ' ', __VA_ARGS__); \
```

Comment:
Since you removed `endl`, I would add a `std::err.flush()` to be sure that the output is printed.

`cerr` should not be buffered, but it is better to be on the safe side if there is no performance penalty as in this case.